server/game/cards/03_TWI/events/DeathByDroids.ts

```diff
@@ -0,0 +1,29 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { WildcardCardType, WildcardZoneName } from '../../../core/Constants';
+
+export default class DeathByDroids extends EventCard {
+    protected override getImplementationId () {
+        return {
+            id: '0968965258',
+            internalName: 'death-by-droids',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.setEventAbility({
+            title: 'Defeat a unit that costs 3 or less. Create 2 Battle Droid tokens',
+            immediateEffect: AbilityHelper.immediateEffects.simultaneous([
+                AbilityHelper.immediateEffects.selectCard({
+                    cardTypeFilter: WildcardCardType.Unit,
+                    zoneFilter: WildcardZoneName.AnyArena,
+                    cardCondition: (card) => card.isUnit() && card.cost <= 3,
+                    innerSystem: AbilityHelper.immediateEffects.defeat(),
+                }),
+                AbilityHelper.immediateEffects.createBattleDroid((context) => ({ target: context.source.controller, amount: 2 })),
```

Comment:
`context.source.controller` is the default so we can just remove the `(context)` handler and do `{ amount: 2 }`. Same for Grievous Reassembly

Reply:
I don't know why, but if I don't specify an explicit target, the test cases fail. It seems like this only happens when it is used in an `immediateEffects.simultaneous()` clause, such as in `CreativeThinking`.

Reply:
Ahh sorry, that's my mistake. I forgot that this is not being used inside of `targetResolver` or `selectCard` so you need to provide the target explicitly.